pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -59,19 +66,54 @@ type StateComputeSpec struct {
 	RenameEnvVar map[string]string `json:"rename,omitempty"`
 }
 
-func generateSecretEnvVarsResource(stateResource *orch_v1.StateResource, computeSpec *StateComputeSpec, dependencyReferences []smith_v1.Reference) (smith_v1.Resource, string, error) {
+func generateSecretResource(compute voyager.ResourceName, envVars map[string]string, dependencyReferences []smith_v1.Reference) (smith_v1.Resource, error) {
+	objectName := wiringutil.MetaNameWithPostfix(compute, secretPluginNamePostfix)
+
+	// if this is for pods then we just have data
+	envVarJSONString, err := json.Marshal(map[string]map[string]string{
+		inputParameterEnvVarName: envVars,
+	})
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+
+	secretPluginSpec, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&secretplugin.Spec{
+		Data: map[string][]byte{
+			envVarOutputSecretKey: envVarJSONString,
+		},
+	})
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+
+	instanceResource := smith_v1.Resource{
+		Name:       smith_v1.ResourceName(objectName),
```

Comment:
he-he, no. This should be `wiringutil.ResourceNameWithPostfix(compute, secretPluginNamePostfix)`. Most of the time `smith_v1.ResourceName()` or `voyager.ResourceName()` type casts are a sign of something fishy going on.